content/cookbook_wrapper.md

```diff
@@ -0,0 +1,145 @@
++++
+title = "Wrapper Cookbooks"
+description = "The cookbook wrapper pattern"
+draft = false
+
+gh_repo = "chef-web-docs"
+product = ["client", "server", "workstation"]
+
+[menu]
+  [menu.infra]
+    title = "Wrapper Cookbooks"
+    identifier = "chef_infra/cookbook_reference/cookbook-wrapper.md About Wrapper Cookbooks"
+    parent = "chef_infra/cookbook_reference"
+    weight = 10
++++
+
+A wrapper cookbook is used to define project- and/or organization-specific requirements around a community cookbook.
+
+Chef recommends using a wrapper cookbook to specify attributes that are unique to your organization.
+
+For example:
+
+```ruby
+default['supermarket_omnibus']['chef_server_url'] = 'https://chefserver.mycompany.com:443'
+default['supermarket_omnibus']['chef_oauth2_app_id'] = '14dfcf186221781cff51eedd5ac1616'
+default['supermarket_omnibus']['chef_oauth2_secret'] = 'a49402219627cfa6318d58b13e90aca'
+default['supermarket_omnibus']['chef_oauth2_verify_ssl'] = false
+default['supermarket_omnibus']['fqdn'] = 'supermarket.mycompany.com'
+```
+
+## Create a Wrapper
+
+Generate a new cookbook using the `chef` command from your workstation:
+
+1. Generate the cookbook:
+
+    ```bash
+    chef generate cookbook my_supermarket_wrapper
+    ```
+
+2. Change directories into that cookbook:
+
+    ```bash
+    cd my_supermarket_wrapper
+    ```
+
+## Define Wrapper Dependencies
+
+3. Defines the wrapper cookbook's dependency on the `SOMETHING` cookbook. Open the metadata.rb file of the new cookbook, and then add the following line:
+
+    ```ruby
+    depends 'SOMETHING'
+    ```
+
+4. Save and close the metadata.rb file.
+
+5. Open the `/recipes/default.rb` recipe located within the new cookbook and add the following content:
+
+    ```ruby
+    include_recipe 'SOMETHING'
+    ```
+
+    This ensures that the `default.rb` file in the `SOMETHING` is run.
+
+## Define Attributes
+
+Define the attributes for the Chef Supermarket installation and how it connects to the Chef Infra Server. One approach is to hard-code attributes in the wrapper cookbook's `default.rb` recipe. A better approach is to place these attributes in a data bag, and then reference them from the recipe. For example, the data bag could be named `apps` and then a data bag item within the data bag could be named `supermarket`.
```

Comment:
```suggestion
Define the attributes for the Chef Supermarket installation and how it connects to the Chef Infra Server. One approach is to hard-code attributes in the wrapper cookbook's `default.rb` recipe.
```

Data bags are *fine*, but I wouldn't say they're the best way of doing this as they're not version controlled.